# Generated by Django 5.1.7 on 2025-03-29 16:27

import django.core.validators
import django.db.models.deletion
import faults.models
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('faults', '0001_initial'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.RemoveField(
            model_name='faultcase',
            name='images',
        ),
        migrations.RemoveField(
            model_name='faultcase',
            name='notes',
        ),
        migrations.AlterField(
            model_name='faultcase',
            name='reported_by',
            field=models.ForeignKey(help_text='Technician or Manager who reported the fault', limit_choices_to={'role__in': ['TECH', 'MANAGER']}, on_delete=django.db.models.deletion.PROTECT, related_name='reported_faults', to=settings.AUTH_USER_MODEL),
        ),
        migrations.CreateModel(
            name='FaultImage',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('image', models.ImageField(help_text='An image showing the fault', upload_to=faults.models.fault_image_upload_path, validators=[django.core.validators.FileExtensionValidator(['jpg', 'jpeg', 'png'])])),
                ('uploaded_at', models.DateTimeField(auto_now_add=True)),
                ('fault_case', models.ForeignKey(help_text='Fault case related to this image', on_delete=django.db.models.deletion.CASCADE, related_name='images', to='faults.faultcase')),
            ],
        ),
        migrations.CreateModel(
            name='FaultNote',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('content', models.TextField(help_text='Content of the note')),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('author', models.ForeignKey(help_text='User who added the note', on_delete=django.db.models.deletion.PROTECT, to=settings.AUTH_USER_MODEL)),
                ('fault_case', models.ForeignKey(help_text='The fault this note is related to', on_delete=django.db.models.deletion.CASCADE, related_name='notes', to='faults.faultcase')),
            ],
        ),
        migrations.CreateModel(
            name='FaultNoteImage',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('image', models.ImageField(help_text='Image associated with the note', upload_to=faults.models.fault_note_image_upload_path, validators=[django.core.validators.FileExtensionValidator(['jpg', 'jpeg', 'png'])])),
                ('uploaded_at', models.DateTimeField(auto_now_add=True)),
                ('note', models.ForeignKey(help_text='Note related to this image', on_delete=django.db.models.deletion.CASCADE, related_name='images', to='faults.faultnote')),
            ],
        ),
    ]
